src/server.ts

```diff
@@ -16,16 +18,25 @@ import Cookies from "universal-cookie";
 const server: ApolloServer = new ApolloServer({
   typeDefs,
   resolvers,
-  context: ({ req }): object => {
-    // to get token from cookies
+  context: async ({ req }): Promise<object> => {
     const cookies = new Cookies(req && req.headers.cookie);
-    const cookieToken = cookies.get("RCTC_USER");
-    // token for backwards compatibility
+    const cookieToken: string = cookies.get("RCTC_USER");
     const fallbackToken = req && req.headers.authorization;
-
+    const token: string = cookieToken ? cookieToken : fallbackToken;
+    const { user, priviledge } =
+      (await resolveUserUsingJWT(prisma, token)) || {};
+    console.log(user);
     return {
       prisma,
-      token: cookieToken ? cookieToken : fallbackToken,
+      token,
+      auth: {
+        user: priviledge === "user" ? user : null,
+        isAuthenticated: priviledge === "user" && user !== null,
+        isAdmin: priviledge === "admin",
+        admin: priviledge === "admin" ? user : null,
+        isAdminAuthenticated:
+          priviledge === "admin" && user !== null ? user : null,
```

Comment:
`isAdminAuthenticated` should be a `boolean` right? which should be just `privilege === 'admin' && user !== null`. Also, I feel some of these fields a little redundant? like maybe `isAuthenticated` and `isAdminAuthenticated` can be combined, basically what we need to achieve is to check if a certain http request is authenticated or not, is it from a `user` or an `admin`, and which `user` or `admin`

Reply:
Oh yes, my bad

Reply:
I think having a `isAdminAuthenticated` is helpful, even though it seems a little redundant right now.

We avoid having to check if `user===null` and so on. We just check if admin is authenticated then use the admin object without worrying about whether its null, or whether user is not null etc.